### ISSUE TITLE
[0.71] Web Prop update caused unexpected breaking changes

### ISSUE THREAD

**Issue by Titozzz** [facebook/react-native#36229]
### Description

Following the work from https://github.com/facebook/react-native/issues/34424, there was some (unexpected?) breaking changes to `<TextInput />`.

Before there were 2 props:
- autoComplete => android (name-family, name-given, etc)
- textContentType => iOS (familyName, givenName, etc)

Now due to the changes, there is still 2 props:
- autoComplete => android & iOS
- textContentType => iOS

But the new implementation looks like this:

There is a new universal set of keys (some of them identical to the old one, some news)
When you set autoComplete:
- [android](https://github.com/facebook/react-native/blob/0e30c7b85321dda42783c6ea09b1955d006295e7/Libraries/Components/TextInput/TextInput.js#L1656): check if name is in the new set of keys, if so [map](https://github.com/facebook/react-native/blob/0e30c7b85321dda42783c6ea09b1955d006295e7/Libraries/Components/TextInput/TextInput.js#L1551) to the android value, else pass it directly to android (compatibility with old keys)
- [iOS](https://github.com/facebook/react-native/blob/0e30c7b85321dda42783c6ea09b1955d006295e7/Libraries/Components/TextInput/TextInput.js#L1663): check if the name is in the set of keys, if so [map](https://github.com/facebook/react-native/blob/0e30c7b85321dda42783c6ea09b1955d006295e7/Libraries/Components/TextInput/TextInput.js#L1585) to the iOS value, else if textContentType is defined, use it.

What's breaking is that for example if you used autoComplete = "A" and textContentType="B", on 0.71 it will be treated as textContentType="A" on 0.71

How we could have mitigated this:
-> Deprecate and set warnings on textContentType but still use it first in the implementation so that user migrate without breaking their apps

### Version

0.71

### Output of `npx react-native info`

/

### Steps to reproduce

Use `TextInput` with autoComplete and textContentType

### Snack, code example, screenshot, or link to a repository

/

**Comment by gabrieldonadel** [comment]
cc: @necolas

**Comment by necolas** [comment]
>  if you used autoComplete = "A" and textContentType="B", on 0.71 it will be treated as textContentType="A" on 0.71

If you use autoComplete = "A" it should be mapped to whatever the corresponding value of textContentType should be, which isn't necessarily "A". Hard to say anything more without a concrete example of the issue you're experiencing

**Comment by NickGerleman** [comment]
If both props were here before, and we repurposed one of them without documenting it as a breaking change, I think we messed up here.

I.e. the web prop precedence should only come into effect for someone using a newly introduced prop.

**Comment by NickGerleman** [comment]
It seems like this change effectively breaks any previous xplat usage of the prop, so we should patch this imo, and make the break later if we want to make it.

**Comment by necolas** [comment]
We added values to `autoComplete` and automatically map them to corresponding iOS values, if the `autoComplete` value is not recognized or not a web value, we use the provided iOS value. The OP needs to explain what specific values are causing issues.

**Comment by NickGerleman** [comment]
It seems like this is the scenario:

> What's breaking is that for example if you used autoComplete = "A" and textContentType="B", on 0.71 it will be treated as textContentType="A" on 0.71

So, if Android and iOS disagreed before. Which is, more scoped, but still technically breaking.

**Comment by necolas** [comment]
Yes but `A` is not a specific value. In what scenarios does the same text input need to have completely different autocomplete values across platforms?

**Comment by Titozzz** [comment]
An example with cc-number, if we have an input that can take credit card information or bank account numbers.
On android, setting cc-number will only impact the suggestions but we do not want to use it on iOS and explicitly set it to none because if you use cc-number on iOS, the keyboard changes and restrains the user from using letters, only numbers.

There you have a regression that was unexpected!

**Comment by Titozzz** [comment]
While I understand this is niche and easily fixable dev side, I'm wondering why this approach was chosen, since the non-breaking alternative strategy (using the textContentType first and warn) seemed easier to me (and would also create discovery for the new capabilities)

**Comment by necolas** [comment]
> the keyboard changes and restrains the user from using letter

When do you need to input letters for a credit card number?

> I'm wondering why this approach was chosen

It's consistent with what we did for dozens of other props and styles.

**Comment by rickhanlonii** [comment]
@necolas interesting, can you list all the props we've done this for?

**Comment by necolas** [comment]
> Bank account have letters in them!

Bank account should not use `cc-number` as that is designed only got credit card numbers, which is why the native keyboard limits it to only numbers.

There is an open HTML issue for adding an autocomplete type for bank accounts, but in the meantime you are expected to use free text entry. On web you can also pattern match to limit to numbers and letters, but React Native doesn't have support for declarative patterns.

https://github.com/whatwg/html/issues/1448

**Comment by Titozzz** [comment]
@necolas I know, but it was a shared input for two data types, that's probably a bad idea but that's what is was. And so have autocomplete for half the data was better than nothing I guess. 🙄 

Once again this is just a silly example revealing something that happened. I don't really want to slow down the effort to migrate to universal props nor argue about the pertinence of old code from our codebase.

The only goal was to show some this was a breaking behaviour while offering an alternative  😄

**Comment by lunaleaps** [comment]
Hey guys for an update, @rickhanlonii, @necolas and I met up to talk about this and the general strategy going forward. 

For this case particularly, I am going to: 
* put out a fix that makes `textContentType` take precedence on iOS and make a patch for 0.71 and ensure its in 0.72
* Update our documentation for `textContentType` that it's just an alias for `autoComplete` but not to use both.  

In our discussion we would like to address this one-off and don't see this as a wide-spread issue (as-in, it likely isn't an issue for other prop aliases we've added). In future efforts around conformance we're going to be more intentional in making block changes behind gating flags so we don't get these paper-cuts for un-intentional adoption. 

cc @NickGerleman

**Comment by Titozzz** [comment]
Thanks

**Comment by donatocayurin** [comment]
Thank you❤️🙏

**Comment by donatocayurin** [comment]
❤️🙏❤️🙏
